datadog/resource_datadog_dashboard.go

```diff
@@ -4669,6 +4783,45 @@ func buildTerraformWidgetConditionalFormat(datadogWidgetConditionalFormat *[]dat
 	return &terraformWidgetConditionalFormat
 }
 
+// Widget Custom Link helpers
+
+func getWidgetCustomLinkSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"label": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"link": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+	}
+}
+func buildDatadogWidgetCustomLinks(terraformWidgetCustomLinks *[]interface{}) *[]datadogV1.WidgetCustomLink {
+	datadogWidgetCustomLinks := make([]datadogV1.WidgetCustomLink, len(*terraformWidgetCustomLinks))
+	for i, customLink := range *terraformWidgetCustomLinks {
+		terraformCustomLink := customLink.(map[string]interface{})
+		datadogWidgetCustomLink := datadogV1.WidgetCustomLink{
+			Label: terraformCustomLink["label"].(string),
+			Link:  terraformCustomLink["link"].(string),
+		}
+		datadogWidgetCustomLinks[i] = datadogWidgetCustomLink
+	}
+	return &datadogWidgetCustomLinks
+}
+func buildTerraformWidgetCustomLinks(datadogWidgetCustomLinks *[]datadogV1.WidgetCustomLink) *[]map[string]string {
+	terraformWidgetCustomLinks := make([]map[string]string, len(*datadogWidgetCustomLinks))
+	for i, datadogWidget := range *datadogWidgetCustomLinks {
```

Comment:
nit: I would rename `datadogWidget` to datadogWidgetCustomLink` since you're iterating over `custom_links` here